### PR TITLE
Task 8 (Docker. AWS Elastic Beanstalk)

### PR DESCRIPTION
**Pull request by Ilya-Valasiuk**
# Frontend:
- https://d3gc8aipmhoxz.cloudfront.net/ (note: It was used self-assigned certificate.)

# Backend:
- https://ilya-valasiuk-cart-api-dev.us-east-1.elasticbeanstalk.com/

# EVALUATION CRITERIA:
- [x] 1 - Dockerfile is prepared, image is building. Image size is minimised to be less than 500 MB.
- [x] 2 - Dockerfile is optimized. Files that change more often and commands that depend on them should be included later, files and commands that change less should be at the top.
- [x] 3 - Folders are added to .dockerignore, with explanations. At least 2 big directories should be excluded from build context. Elastic Beanstalk application is initialized.
- [x] 4 - Environment is created and the app is deployed to the AWS cloud. You must provide a link to your GitHub repo with Cart API service or PR with created Dockerfile and related configurations.
- [x] 5 - FE application is updated with Cart API endpoint. You must provide a PR with updates in your FE repository and OPTIONALLY link to deployed front-end app which makes proper API calls to your Cart service.

# Docker image:
image

# Note:
In `.gitignore` you can see that I have `private.txt`. This file contains my self-assigned private certificate. It was used during the app build.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+FROM node:15.3.0-alpine3.10 as base
```
Comment (Ivanko5417) [inline Dockerfile:1]: I don't think it's a good idea to use non-LTS node version

**Comment by ArtiFisher** [comment]
Seems like there is some certificate issue with back-end link:
https://ilya-valasiuk-cart-api-dev.us-east-1.elasticbeanstalk.com/

**Comment by Ilya-Valasiuk** [comment]
@ArtiFisher It's using self-assigned certificate. You need to add this to trusted or use FF in incognito mode

File: Dockerfile (hunk 1 above)
Comment (Ilya-Valasiuk) [reply Dockerfile:1]: why not? it's just a playground

**Comment by ArtiFisher** [comment]
Adding unknown certificates to trusted seems like a bad idea. And it's not opening in incognito mode too. Besides it's surely not something that user of a store would do :)

**Comment by Ilya-Valasiuk** [comment]
@ArtiFisher Well, as I understand here we are trying to create improve our knowledges in AWS.
If you will open FF (Firefox) in incognito mode - 
image
you can see that you have possibility to take risks and continue with that API. You will see that everything working as expected. 
image

The problem here is I tried to create self-assigned certificate and to override `ngix` server so it will response with this certificate. And if you remember lector during the session said that creating self-assigned certificate is another challenge in this task. 

To address to your point that it's not something that real user will do - of course you right. BUT it's not a real shop and here we trying to get knowledges about AWS. When you don't try to get in changes that was done in this PR and just mark as 4 - it's not a fair play. 

As you can see I have added notice in PR description that you will faced with that issues.

**Comment by ArtiFisher** [comment]
Seems like only firefox allows it. I tested it in other browsers. Also you marked front-end link with a note about certificate, not back-end.
Anyway I checked that it works in firefox, so will change mark to 5.

**Pull request merged**